File: jajo-backend/src/main/java/jajobackend/model/Product.java
package jajobackend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.ToString;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Long id;

    @Column (nullable = false, length = 100)
    private String name;

    @Column (nullable = false)
    private Integer price;

    @Column (nullable = false)
    private Integer cost;

    @Column (nullable = false)
    private Long hierarchy;

    //One to Many
    @OneToMany (mappedBy = "product")
    @JsonIgnore
    @ToString.Exclude
    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    private List<Count> counts;

    public Product() {}
}

File: jajo-frontend/src/app/components/model/product.ts
export interface GetProduct {
  id: number;
  name: string;
  price: number;
  cost: number;
  hierarchy: number;
  name_readonly: boolean;
  price_readonly: boolean;
  cost_readonly: boolean;
}

export interface PostProduct {
  id: number;
  name: string;
  price: number;
  cost: number;
  hierarchy: number;
}
